work/views.py

```diff
@@ -953,38 +1025,100 @@ def default_order(self):
+            'manifestations_searchable': [
+                'manif_set__manifestation_type',
+                'manif_set__postage_marks',
+                'manif_set__id_number_or_shelfmark',
+                'manif_set__printed_edition_details',
+                'manif_set__manifestation_incipit',
+                'manif_set__manifestation_excipit',
+                'manif_set__cofkmanifinstmap_set__inst__institution_name',
+                'manif_set__manif_from_set__manif_to__id_number_or_shelfmark',
+                'manif_set__manif_to_set__manif_from__id_number_or_shelfmark',
+            ],
+            'language_of_work': [
+                'language_set__language_code__code_639_3',
+                'language_set__language_code__code_639_1',
+                'language_set__language_code__language_name',
+            ],
+            'original_catalogue': [
+                'original_catalogue__catalogue_code',
+                'original_catalogue__catalogue_name',
+            ],
+            'images': [
+                'manif_set__images__image_filename',
+            ],
+        }
 
         queries.extend(
-            query_utils.create_queries_by_lookup_field(request_data, self.search_fields, search_fields_maps)
+            query_utils.create_queries_by_lookup_field(
+                request_data, self.search_fields,
+                search_fields_maps=search_fields_maps,
+                search_fields_fn_maps={
+                    'creators_searchable': create_lookup_fn_by_person([REL_TYPE_CREATED]),
+                    'notes_on_authors': create_lookup_fn_by_comment([REL_TYPE_COMMENT_AUTHOR]),
+                    'places_from_searchable': create_lookup_fn_by_location([REL_TYPE_WAS_SENT_FROM]),
+                    'addressees_searchable': create_lookup_fn_by_person([REL_TYPE_WAS_ADDRESSED_TO]),
+                    'places_to_searchable': create_lookup_fn_by_location([REL_TYPE_WAS_SENT_TO]),
+                    'sender_or_recipient': create_lookup_fn_by_person([REL_TYPE_CREATED, REL_TYPE_WAS_ADDRESSED_TO]),
+                    'origin_or_destination': create_lookup_fn_by_location(
+                        [REL_TYPE_WAS_SENT_FROM, REL_TYPE_WAS_SENT_TO]),
+                    'related_resources': create_lookup_fn_by_resource([REL_TYPE_IS_RELATED_TO]),
+                    'general_notes': create_lookup_fn_by_comment([REL_TYPE_COMMENT_REFERS_TO]),
+                    'people_mentioned': create_lookup_fn_by_person([REL_TYPE_MENTION]),
+                    'flags': lookup_fn_flags,
+                })
         )
-        return self.create_queryset_by_queries(CofkUnionQueryableWork, queries, sort_by=sort_by).distinct()
+        return self.create_queryset_by_queries(CofkUnionWork, queries, sort_by=sort_by)
+
+    def create_queryset_by_queries(self, model_class: Type[models.Model], queries: Iterable[Q],
+                                   sort_by=None):
+        queryset = model_class.objects.filter()
```

Comment:
Should this be a call to filter without any parameters?

Reply:
yes, it will throw exception if remove `.filter()`.
but I'm not sure will `.all()` trigger DB and load everything to ram when calling it.
so, empty `.filter()` seem fine to create queryset for the `for loop`